MayssaraOmar/assignment3/Word search/src/Word search.cpp

```diff
@@ -0,0 +1,48 @@
+#include <map>
+#include "word search.h"
+using namespace std;
+
+void get_words_from_grid(vector<vector<char> > &grid, int row, int col,
+		string word, map<pair<int, int>, bool> &vis,
+		map<pair<pair<int, int>, string>, bool> &memo, dictionary &dic,
+		unordered_set<string> &words) {
```

Comment:
Style: A common convention is to make output parameters (parameters that are outputs or modified by a function) pointers rather than references.